src/java-runtime/java/mono/android/MonoPackageManager.java

```diff
@@ -55,6 +55,7 @@ public static void LoadApplication (Context context, ApplicationInfo runtimePack
 				// needed in the latest Android versions but is required in at least
 				// API 16 and since there's no inherent negative effect of doing it,
 				// we can do it unconditionally.
+				System.loadLibrary("monosgen-2.0");
```

Comment:
It would be cool if you could try to load Mono from `externalDir` first.  That way @brendanzagaeski and I can still drop in a hacked up `libmonosgen-2.0` if we want to triage some Mono regression.

Reply:
We can't do it unconditionally. The code has to work for all apps and access to `externalDir` needs an Android permission to be granted (I think that holds true even for debug mode apps).